### ISSUE TITLE
Could allow for sites to be bound to switches?

### ISSUE THREAD

**Issue by kitsunde** [YPlan/gargoyle#74]
We have several multi-tennant sites and currently we're doing something pretty awkward with waffle where we postfix the site like (`feature_video_sg`, `feature_video_nz`) and then do an awkward lookup.

**Comment by adamchainz** [comment]
What about a site condition, where you turn a switch on only for specific sites?

**Comment by adamchainz** [comment]
Also I presume you're talking about Django's Site framework

**Comment by kitsunde** [comment]
Yeah we're using the standard Django Site framework. Ideally we would bind the switches m2m to the `sites.Site` model.

**Comment by adamchainz** [comment]
You should be able to create a `ConditionSet` based on `RequestConditionSet` that does this.

See `IPAddressConditionSet` for a sample implementation that looks at the IP of the request - you could easily adapt it to look at `request.get_current_site()`. `IPAddressConditionSetTests` are also a pretty good guide on how to test.

If you start and open a PR I'll help with code review and completing anything that turns out to be difficult. Of course you can implement it in your own project's code base right now if you need it right now :)
